src/cmd/cloud.js

```diff
@@ -378,16 +412,31 @@ module.exports = class CloudCommand extends CLICommandBase {
 					throw normalizedApiError(resp);
 				}
 			})
-			.catch(err => {
-				throw normalizedApiError(err);
-			})
 			.then((sizeInfo) => {
-				if (sizeInfo){
+				respSizeInfo = sizeInfo;
+				if (assets) {
+					return new BundleCommands()._generateBundle({ assetsList: assets, appBinary: filename, bundleFilename: bundleFilename });
+				}
+			})
+			.then((bundle) => {
+				if (respSizeInfo){
 					this.ui.stdout.write(`Memory use:${os.EOL}`);
-					this.ui.stdout.write(`${sizeInfo}${os.EOL}`);
+					this.ui.stdout.write(`${respSizeInfo}${os.EOL}`);
+				}
+				if (bundle) {
+					if (fs.existsSync(filename)){
+						fs.unlinkSync(filename);
+					}
+					// TODO: Verify that the bundle is created with the correct files
+					this.ui.stdout.write(`Compile succeeded and bundle created.${os.EOL}`);
+					this.ui.stdout.write(`Saved bundle to: ${path.resolve(bundleFilename)}${os.EOL}`);
+				} else {
+					this.ui.stdout.write(`Compile succeeded.${os.EOL}`);
+					this.ui.stdout.write(`Saved firmware to: ${path.resolve(filename)}${os.EOL}`);
 				}
-				this.ui.stdout.write(`Compile succeeded.${os.EOL}`);
-				this.ui.stdout.write(`Saved firmware to: ${path.resolve(filename)}${os.EOL}`);
+			})
+			.catch((err) => {
+				throw normalizedApiError(err);
```

Comment:
normalizedApiError should only be called after a Particle API call. It's not suitable for a general catch clause after a long function. What issue did you encounter that you are trying to fix by adding this?

Reply:
Thank you for highlighting. The exact scenario is this e2e test which will not return the error https://github.com/particle-iot/particle-cli/blob/master/test/e2e/compile.e2e.js#L502C29-L525

so I pulled the catch block down to the very ending. I am guessing I will have to refactor this a bit 🤔

Reply:
Refactored to use async/await so I believe my concern here doesn't hold true anymore :)